src/mpid/common/shm/mpidu_shm_obj.h

```diff
@@ -0,0 +1,19 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+#ifndef MPIDU_SHM_OBJ_H_INCLUDED
+#define MPIDU_SHM_OBJ_H_INCLUDED
+
+typedef enum MPIDU_shm_obj {
+    MPIDU_SHM_OBJ__NONE = -1,
+    MPIDU_SHM_OBJ__FASTBOXES = 0,
+    MPIDU_SHM_OBJ__CELLS,
+    MPIDU_SHM_OBJ__COPYBUFS,
+    MPIDU_SHM_OBJ__WIN,
+    MPIDU_SHM_OBJ__NUM
+} MPIDU_shm_obj_t;
```

Comment:
The memory objects (I'd call these `buffer types`, instead) are different from memory types.  These two should be split into different commits.  Memory types are usable even without explicit buffer types, i.e., we can have all buffers be allocated on DRAM or MCDRAM or something else.  Buffer types only allow us to have some buffers on one memory type and other buffers on a different memory type.

Reply:
Makes sense, I will split the two in different commits.